frontend/packages/operator-lifecycle-manager/src/components/modals/disable-default-source-modal.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { k8sPatch, K8sKind } from '@console/internal/module/k8s';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory/modal';
+import { YellowExclamationTriangleIcon } from '@console/shared';
+import { withHandlePromise, HandlePromiseProps } from '@console/internal/components/utils';
+import { OperatorHubKind } from '../operator-hub';
+
+const DisableDefaultSourceModal: React.FC<DisableSourceModalProps> = ({
+  kind,
+  operatorHub,
+  sourceName,
+  close,
+  cancel,
+  inProgress,
+  errorMessage,
+  handlePromise,
+}) => {
+  const submit = React.useCallback(
+    (event) => {
+      event.preventDefault();
+      const currentSources = _.get(operatorHub, 'spec.sources', []);
+      const patch = [
+        {
+          op: _.isEmpty(currentSources) ? 'add' : 'replace',
+          path: '/spec/sources',
+          value: [
+            ..._.filter(currentSources, (source) => source.name !== sourceName),
+            {
+              name: sourceName,
+              disabled: true,
+            },
```

Comment:
Do we care that this is reordering the sources? (I'm not sure we do, just asking :)